src/Aspire.Hosting/DistributedApplicationLifecycle.cs

```diff
@@ -21,13 +21,13 @@ public Task StartAsync(CancellationToken cancellationToken)
 
     public Task StartedAsync(CancellationToken cancellationToken)
     {
-        if (distributedApplicationOptions.DashboardEnabled && configuration["AppHost:BrowserToken"] is { Length: > 0 } browserToken)
-        {
-            LoggingHelpers.WriteDashboardUrl(logger, configuration["ASPNETCORE_URLS"], browserToken);
-        }
-
         if (executionContext.IsRunMode)
         {
+            if (distributedApplicationOptions.DashboardEnabled && configuration["AppHost:BrowserToken"] is { Length: > 0 } browserToken)
```

Comment:
Not a bad idea to omit this output if the browser token is missing 👍

But you might also wish to omit the output if IsPublishMode == true (look at the `DistributedApplicationExecutionContext` for this. If someone modifies their launchprofile to pass in the `--publisher manifest` argument then they'll produce a manifest but this output will still be output, because VS would have injected the browser token (based on my understanding of how the browser token injection mechanism is going to work).

Reply:
Approving now since that is more of an edge case (we certainly experience it in our repo but we are probably outliers).

Reply:
It's nested inside an `IsRunMode` check. If there is a publisher value then won't that be false?

Reply:
It isn't included when run from VS with `"commandLineArgs": "--publisher manifest --output-path test.json"` in launch profile:

![image](https://github.com/dotnet/aspire/assets/303201/2beb0184-ecca-4bfb-ba33-1723529f6edd)